plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionPageGeneral.java

```diff
@@ -562,8 +563,16 @@ public void saveSettings(DBPDataSourceContainer dataSource) {
             return;
         }
         final DBPConnectionConfiguration confConfig = dataSource.getConnectionConfiguration();
-
-        String name = connectionNameChanged ? generateConnectionName(getWizard().getPageSettings(), connectionNameText.getText()) : generateConnectionName(getWizard().getPageSettings(), ModelPreferences.getPreferences().getString(ModelPreferences.DEFAULT_CONNECTION_NAME_PATTERN));
+        String name;
+        if (dataSourceDescriptor == null) {
+            name = connectionNameChanged
+                    ? generateConnectionName(getWizard().getPageSettings(), connectionNameText.getText()) :
+                    generateConnectionName(getWizard().getPageSettings(), ModelPreferences.getPreferences()
+                            .getString(ModelPreferences.DEFAULT_CONNECTION_NAME_PATTERN));
+        } else {
+            name = connectionNameChanged ? generateConnectionName(getWizard().getPageSettings(),
+                    connectionNameText.getText()) : dataSourceDescriptor.getName();
+        }
```

Comment:
1. Please don't mix `if` statements with ternary operator expressions.
2. Continuation indentation must be four spaces, not 8.